code/src/components/LikeBtn.js

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable no-underscore-dangle */
+import React, { useState } from 'react';
+
+const LikeBtn = ({ event, setNewLike }) => {
+  const [liked, setLiked] = useState(false);
+
+  // the function that enables the likes-count to update in the App-component
+  const handleNewLikeChange = () => {
+    setNewLike(true)
+  }
+
+  // The function that registers the users like to the database, it also
+  // triggers the state-hook that makes the button change color
+  const onButtonClick = (event2) => {
+    event2.preventDefault();
+
+    if (liked === false) {
+      const options = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: ''
+      }
+
+      fetch(`https://happy-thoughts-technigo.herokuapp.com/thoughts/${event._id}/like`, options)
+        .then((result) => result.json())
+        .catch((error) => console.error(error))
+        .finally(() => {
+          setLiked(true)
+          handleNewLikeChange()
```

Comment:
Very good comment! I had a second of doubt about the naming and then forgot about it, wondering if it would be confusing. One is for the registration of the "like"-post to the database(and as you observed it can only be clicked once) and the other is for the site-update.